thorclient/http/client.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/vechain/thor/v2/api/accounts"
+	"github.com/vechain/thor/v2/api/blocks"
+	"github.com/vechain/thor/v2/api/events"
+	"github.com/vechain/thor/v2/api/node"
+	"github.com/vechain/thor/v2/api/transactions"
+	"github.com/vechain/thor/v2/api/transfers"
+	"github.com/vechain/thor/v2/thor"
+	"github.com/vechain/thor/v2/thorclient/common"
+)
+
+type Client struct {
+	url string
+	c   *http.Client
+}
+
+func NewClient(url string) *Client {
+	return &Client{
+		url: url,
+		c:   &http.Client{},
+	}
+}
+
+func (c *Client) GetTransactionReceipt(txID *thor.Bytes32) (*transactions.Receipt, error) {
+	body, err := c.httpGET(c.url + "/transactions/" + txID.String() + "/receipt")
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch receipt - %w", err)
+	}
+
+	if len(body) == 0 || bytes.Equal(bytes.TrimSpace(body), []byte("null")) {
+		return nil, common.NotFoundErr
+	}
+
+	var receipt transactions.Receipt
+	if err = json.Unmarshal(body, &receipt); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall receipt - %w", err)
+	}
+
+	return &receipt, nil
+}
+
+func (c *Client) InspectClauses(calldata *accounts.BatchCallData) ([]*accounts.CallResult, error) {
+	body, err := c.httpPOST(c.url+"/accounts/*", calldata)
+	if err != nil {
+		return nil, fmt.Errorf("unable to request inspect clauses - %w", err)
+	}
+
+	var inspectionRes []*accounts.CallResult
+	if err = json.Unmarshal(body, &inspectionRes); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall inspection - %w", err)
+	}
+
+	return inspectionRes, nil
+}
+
+func (c *Client) SendTransaction(obj *transactions.RawTx) (*common.TxSendResult, error) {
+	body, err := c.httpPOST(c.url+"/transactions", obj)
+	if err != nil {
+		return nil, fmt.Errorf("unable to send raw transaction - %w", err)
+	}
+
+	var txID common.TxSendResult
+	if err = json.Unmarshal(body, &txID); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall inspection - %w", err)
+	}
+
+	return &txID, nil
+}
+
+func (c *Client) GetLogs(eventEndpoint string, req map[string]interface{}) ([]events.FilteredEvent, error) {
```

Comment:
Maybe `GetEvents` so it matches API name?

Also, `Get` gives me an ick since it is a `POST` request  :stuck_out_tongue_closed_eyes:

Maybe `FilterEvents`?

Reply:
Ahaha, that's something I didn't notice